src/hp/doctor.rs

```diff
@@ -0,0 +1,609 @@
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use crate::{
+    hp::{
+        self,
+        derp::{DerpMap, UseIpv4, UseIpv6},
+        key::node::SecretKey,
+        magicsock,
+    },
+    tls,
+    tokio_util::ProgressWriter,
+};
+use anyhow::Context;
+use clap::Subcommand;
+use indicatif::{HumanBytes, MultiProgress, ProgressBar};
+use postcard::experimental::max_size::MaxSize;
+use serde::{Deserialize, Serialize};
+use tokio::{io::AsyncWriteExt, sync};
+
+#[derive(Subcommand, Debug, Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum Commands {
+    Report {
+        #[clap(long, default_value = "derp.iroh.computer")]
+        host_name: String,
+        #[clap(long, default_value_t = 3478)]
+        stun_port: u16,
+    },
+    Accept {
+        /// Our own private key, in hex. If not specified, a random key will be generated.
```

Comment:
Does this allow for reusing the iroh key? would potentially be useful to fallback to the stored one.

Reply:
I added an option local to use the key from the iroh dir. But I don't think it should be the default.

Maybe it should be the default for accept, but not connect? You don't want two nodes with the same peer id to try to connect, so I think random is a good default.